web/crux/src/app/deploy/interceptors/deploy.start.interceptor.ts

```diff
@@ -91,6 +91,30 @@ export default class DeployStartValidationInterceptor implements NestInterceptor
       })
     }
 
+    if (!deployment.protected) {
+      const {
+        query: { ignoreProtected },
+      } = req
+
+      if (!ignoreProtected || ignoreProtected === false) {
+        const otherProtected = await this.prisma.deployment.findFirst({
+          where: {
+            protected: true,
+            nodeId: deployment.nodeId,
+            prefix: deployment.prefix,
+          },
+        })
+
+        if (otherProtected != null) {
```

Comment:
I think this warning should be thrown when:
Rolling versions: `There is another deployment, which is protected on the same node same prefix`
Incremental: `There is another deployment, which is protected on the same node same prefix and it isn't this version's parent or child.`

Reply:
So in case of an incremental deployment I should also check if there's another deployment which is in another version.

Reply:
Yep. The query now looks good!